### PR TITLE
SDI-652 Tighten validation rules on activities API

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityResourceIntTest.kt [hunk 1]
```diff
@@ -72,79 +71,50 @@ class ActivityResourceIntTest : IntegrationTestBase() {
 
   @Nested
   inner class CreateActivity {
-
-    private val createActivityRequest: () -> CreateActivityRequest = {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityResourceIntTest.kt:76]: Most of the tests were using JSON now so switched the remainder for consistency

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityResourceIntTest.kt [hunk 2]
```diff
@@ -72,79 +71,50 @@ class ActivityResourceIntTest : IntegrationTestBase() {
...
     fun `error from pay rate service should return bad request`() {
-      val invalidPayRate = payRateRequest.copy(payBand = "INVALID")
-      webTestClient.post().uri("/activities")
-        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ACTIVITIES")))
-        .body(BodyInserters.fromValue(createActivityRequest().copy(payRates = listOf(invalidPayRate))))
-        .exchange()
-        .expectStatus().isBadRequest
+      val invalidSchedule = validJsonRequest().replace(""""payBand" : "5",""", """"payBand" : "INVALID",""")
+
+      createActivityExpectingBadRequest(invalidSchedule)
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityResourceIntTest.kt:117]: Remove all the noise from webTestClient

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/CreateActivityRequest.kt [hunk 3]
```diff
@@ -41,6 +47,7 @@ data class CreateActivityRequest(
   val programCode: String,
 
   @Schema(description = "Half or Full day (H or F)", required = true, example = "H")
+  @field:Pattern(regexp = "[H|F]")
   val payPerSession: String,
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/CreateActivityRequest.kt:51]: I wonder if we can use the enum type here and let 'spring' do the validation

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**